language/tools/cost-synthesis/src/stack_generator.rs

```diff
@@ -205,12 +207,14 @@ where
                 peek.expect("[Next Integer] Unable to cast peeked stack value to a u8."),
             )
         } else {
-            self.gen.gen_range(0, u8::max_value())
+            self.gen.gen_range(1, 16u8)
         }
     }
 
-    fn next_u64(&mut self, stk: &[Value]) -> u64 {
-        if self.op == Bytecode::Sub && !stk.is_empty() {
+    fn next_u64(&mut self, stk: &[Value], is_padding: bool) -> u64 {
+        if (self.op == Bytecode::Shl || self.op == Bytecode::Shr) && !is_padding {
+            self.gen.gen_range(1, 8u64)
```

Comment:
should this be 64u8? or 64 in general?

Reply:
No. Here we're generating the second argument for shl & shr, and the maximum value allowed depends on the type of the first argument, which is tricky to get here. Therefore a safe choice (max 8) was made. 

Tim is planning to refactor cost-synthesis at a later time so we decided to not bother making the code perfect.

Reply:
Oh wait I misread. But yeah the same logic still applies:
1. Second argument must be less than the number of bits in the first argument, which has a lower bound 8.
2. This function is returning a u64, and the suffix in the literal was intended to make this explicit. Let me see if type inference could figure this out itself.